### PR TITLE
feat: generic engine in the shim library

### PR DESCRIPTION
**Pull request by Mossaka**
This PR implements a generic engine to the shim library. This removes the dependency on `wasmtime::Engine` in the library entirely, and it is up to the implementation to decide what instance of the engine should be.

### CODE REVIEW

File: crates/containerd-shim-wasmtime-v1/src/sandbox/manager.rs [hunk 1]
```diff
@@ -24,33 +24,35 @@ use std::sync::Arc;
...
             phantom: std::marker::PhantomData,
         }
     }
 }
 
-impl<T> Manager for Service<T>
+impl<E: Send + Sync + Clone, T> Manager for Service<E, T>
 where
-    T: Sandbox<Instance = WasiInstance> + 'static,
+    T: Sandbox<E, Instance = WasiInstance> + 'static,
```
Comment (cpuguy83) [inline crates/containerd-shim-wasmtime-v1/src/sandbox/manager.rs:55]: Can `E`'s type constraints be moved into the `where` clause here?

**Review by cpuguy83 — CHANGES_REQUESTED** [review]
Minor cosmetic change, everything else looks good.

File: crates/containerd-shim-wasmtime-v1/src/sandbox/manager.rs (hunk 1 above)
Comment (Mossaka) [reply crates/containerd-shim-wasmtime-v1/src/sandbox/manager.rs:63]: Resolved

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Pull request merged**